z_json_test.go

```diff
@@ -705,6 +705,126 @@ func TestUpdateJSONScalar(t *testing.T) {
 	}
 }
 
+// Converts map with different Go-types to godror.JSON type
+// This is bound to the insert function and executed
+// For each unique go-type in the map, their corresponding JSON types,
+// as stored in the DB, are fetched and compared with their expected values.
+func TestStorageTypes(t *testing.T) {
+	t.Parallel()
+	ctx, cancel := context.WithTimeout(testContext("StorageTypes"),
+		30*time.Second)
+	defer cancel()
+	conn, err := testDb.Conn(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer conn.Close()
+	tbl := "test_objectcollection_jsonmap" + tblSuffix
+	conn.ExecContext(ctx, "DROP TABLE "+tbl)
+	_, err = conn.ExecContext(ctx,
+		"CREATE TABLE "+tbl+" (id NUMBER(6), jdoc JSON)", //nolint:gas
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Logf(" JSON Document table  %q: ", tbl)
+
+	defer testDb.Exec(
+		"DROP TABLE " + tbl, //nolint:gas
+	)
+
+	stmt, err := conn.PrepareContext(ctx,
+		"INSERT INTO "+tbl+" (id, jdoc) VALUES (:1, :2)", //nolint:gas
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer stmt.Close()
+	//        birthdate, err := time.Parse(time.UnixDate, "Wed Feb 25 11:06:39 PST 1990")
+	birthdate := time.Now()
+
+	// Map with different Go-types, which would be stored as JSON in the DB
+	jsmap := map[string]interface{}{
+		"asObject": map[string]interface{}{
+			"asNumber":    godror.Number("12"),
+			"asString":    "Mary",
+			"asTimestamp": birthdate,
+			"asBoolean":   true,
+			"asByte":      []byte{45, 51},
+			"asInt32":     int32(98),
+			"asInt64":     int64(99),
+			"asInt8":      int8(10),
+			"asInt16":     int16(20),
+			"asUint64":    uint64(99),
+			"asFloat64":   float64(98.11),
+			"asFloat32":   float32(97.2),
+		},
+	}
+
+	jsonval := godror.JSONValue{Value: jsmap}
+	if _, err = stmt.ExecContext(ctx, 0, jsonval); err != nil {
+		t.Errorf("%d/1. (%v): %v", 0, jsmap, err)
+
+	}
+
+	rows, err := conn.QueryContext(ctx,
+		"SELECT id,json_value(jdoc,'$.asObject.type()'),json_value(jdoc,'$.asObject.asString.type()'),json_value(jdoc,'$.asObject.asNumber.type()'),json_value(jdoc,'$.asObject.asBoolean.type()'),json_value(jdoc,'$.asObject.asTimestamp.type()'),json_value(jdoc,'$.asObject.asInt32.type()'),json_value(jdoc,'$.asObject.asInt64.type()'),json_value(jdoc,'$.asObject.asInt8.type()'),json_value(jdoc,'$.asObject.asInt16.type()'),json_value(jdoc,'$.asObject.asUint64.type()'),json_value(jdoc,'$.asObject.asFloat64.type()'),json_value(jdoc,'$.asObject.asFloat32.type()'),json_value(jdoc,'$.asObject.asByte.type()') FROM "+tbl+" c ") //nolint:gas
```

Comment:
split with spaces, not just commas.
Generate if possible (from the `jsmap["asObject"]` keys?)